services/system/StagedTx/service/src/lib.rs

```diff
@@ -0,0 +1,440 @@
+
+    struct StagedTxPolicy {
+        service_caller: ServiceCaller,
+    }
+    impl StagedTxPolicy {
+        pub fn new(auth_service: AccountNumber) -> Self {
+            StagedTxPolicy {
+                service_caller: ServiceCaller {
+                    sender: Wrapper::SERVICE,
+                    service: auth_service,
+                },
+            }
+        }
+
+        pub fn accept(&self, id: u32, actor: psibase::AccountNumber) {
+            self.service_caller.call_returns_nothing(
+                MethodNumber::from(auth_action_structs::stagedAccept::ACTION_NAME),
+                auth_action_structs::stagedAccept {
+                    staged_tx_id: id,
+                    actor,
+                },
+            )
+        }
+
+        pub fn reject(&self, id: u32, actor: psibase::AccountNumber) {
+            self.service_caller.call_returns_nothing(
+                MethodNumber::from(auth_action_structs::stagedReject::ACTION_NAME),
+                auth_action_structs::stagedReject {
+                    staged_tx_id: id,
+                    actor,
+                },
+            )
+        }
+    }
+
+    #[derive(Fracpack, Serialize, Deserialize, ToSchema, SimpleObject)]
+    pub struct ActionList {
+        actions: Vec<Action>,
+    }
+
+    #[table(name = "InitTable", index = 0)]
+    #[derive(Serialize, Deserialize, ToSchema, Fracpack)]
+    struct InitRow {}
+    impl InitRow {
+        #[primary_key]
+        fn pk(&self) {}
+    }
+
+    #[table(name = "StagedTxTable", index = 1)]
+    #[derive(Fracpack, Serialize, Deserialize, ToSchema, SimpleObject)]
+    pub struct StagedTx {
+        pub id: u32,
+        pub txid: [u8; 32],
+        pub propose_date: TimePointUSec,
+        pub proposer: AccountNumber,
+        pub action_list: ActionList,
+    }
+    impl StagedTx {
+        #[primary_key]
+        fn by_id(&self) -> u32 {
+            self.id
+        }
+
+        #[secondary_key(1)]
+        fn by_act_sender(&self) -> (AccountNumber, u32) {
+            check(
+                self.action_list.actions.len() > 0,
+                "Staged transaction contains no actions",
+            );
+            (self.action_list.actions[0].sender, self.id)
+        }
+
+        #[secondary_key(2)]
+        fn by_staging_policy(&self) -> (AccountNumber, u32) {
+            check(
+                self.action_list.actions.len() > 0,
+                "Staged transaction contains no actions",
+            );
+            let sender = self.action_list.actions[0].sender;
+            let auth_service =
+                get_auth_service(sender).expect("Action sender account in staged tx is invalid");
+            (auth_service, self.id)
+        }
+
+        fn new(actions: Vec<Action>) -> Self {
+            check(
+                actions.len() > 0,
+                "Staged transaction must contain at least one action",
+            );
+
+            let sender = actions[0].sender;
```

Comment:
Correct, there wasn't, because the `runAs` for the other sender will fail unless that sender's auth service allows runAsOtherReq (which, frankly, I don't know why any auth service would ever allow that).

Reply:
And now there isn't for a second reason, which is that I explicitly restricted proposed staged transactions so that they must all be from the same sender to address the loophole you found.